soc/bouffalolab/common/idle.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2017 Jean-Paul Etienne <fractalclone@gmail.com>
+ * Contributors: 2018 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/toolchain.h>
+#include <zephyr/irq.h>
+#include <zephyr/arch/cpu.h>
+
+#include <zephyr/tracing/tracing.h>
+
+static ALWAYS_INLINE void riscv_idle(unsigned int key)
+{
+	sys_trace_idle();
+	/* unlock interrupts */
+	irq_unlock(key);
```

Comment:
This can probably just leave `CONFIG_ARCH_HAS_CUSTOM_CPU_IDLE` undefined and use the default RISCV implementation in  `arch/riscv/core/cpu_idle.c`, see also #70372